pkg/alertmanager/operator.go

```diff
@@ -543,11 +540,8 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 		return nil
 	}
 
-	logger := c.logger.With("key", key)
 	logDeprecatedFields(logger, am)
 
-	logger.Info("sync alertmanager")
```

Comment:
It's an info log which is why it's printed only when the controller does actual work (e.g. no deletion in progress or pause).